content/blog/2017/2017-02-06-declarative-pipeline-ga.adoc

```diff
@@ -0,0 +1,89 @@
+
+* Retain history of all changes to Pipeline
+* Rollback to a previous pipeline version
+* Review new changes to the pipeline in code review
+* Test new pipeline steps in branches
+* Audit changes to the pipeline
+* Run the same pipeline on a different Jenkins server
+
+= Getting Started with Declarative Pipeline
+We recommend people begin using it for all their Pipeline definitions in Jenkins.
+The plugin has been available for use and testing starting with the 0.1 release that was debuted at
+link:https://www.cloudbees.com/introducing-new-way-define-jenkins-pipelines[Jenkins World]
+in September, it has already been installed over 5,000 times.
+
+If you haven't tried Pipeline or have considered Pipeline in the past we believe
+this new syntax is much more approachable with an easy adoption curve to quickly
+realize all of the benefits of Pipeline as Code. In addition, the predefined structure
+of Declarative makes it possible to create and edit pipelines with a graphical
+user interface (GUI). The Blue Ocean team is actively working on a Pipeline Editor
+that will be included in an upcoming release.
+
+If you have already begun using Pipelines in Jenkins we believe that this new alternative
+syntax can help expand that usage. The original syntax for defining pipelines in Jenkins
+is a Groovy DSL that allows most of the features of full imperative programming.
+This syntax is still fully supported and is now referred to as "Scripted Pipeline Syntax"
+to distinguish it from "Declarative Pipeline Syntax." Both use the same underlying execution
+engine in Jenkins and both will generate the same results in Pipeline Stage View or Blue Ocean
+visualization. All existing pipeline steps, global variables and shared libraries can be used
+in either. You can now create more cookie-cutter pipelines and extend the power of Pipeline to all users regardless of Groovy expertise.
+
+= Declarative Pipeline Features
+
+* link:https://en.wikipedia.org/wiki/Lint_%28software%29[Syntax Checking]
+** Immediate runtime syntax checking with explicit error messages
+** API endpoint for linting Jenkinsfiles
+** CLI command to lint Jenkinsfiles
+* link:https://plugins.jenkins.io/docker-workflow[Docker Pipeline plugin integration]
+** Run all stages in a single container
+** Run each stage in a different container
+* Easy configuration
+** Quickly define parameters for your pipeline
+** Quickly define environment variables and credentials for your pipeline
+** Quickly define options (such as timeout, retry, build discarding) for your pipeline
+* Conditional actions
+** Send notifications or take actions depending upon success or failure
+** Skip stages based on branches, environment, or other Boolean expression
```

Comment:
** Round trip editing with the visual pipeline editor (watch for preview release shortly)